### PR TITLE
[ui] Autodetect links in context.log messages, render clickable links

### PR DESCRIPTION
**Pull request by bengotow**
## Summary & Motivation

This is a fix for #15862. It's largely a port of the autolinker I wrote for Mailspring [here](https://github.com/Foundry376/Mailspring/blob/master/app/src/services/autolinker.ts), which doesn't require external packages and has gotten a handful of bug fixes over the years (MIT-licensed so it's safe to reference.)

The code uses a useEffect to look at the text node after it's rendered in React and adds the `a` tags directly at the DOM level. All the links open in new tabs and have rel=nofollow, noreferrer for safety's sake. It uses useIdleCallback + a DOM treewalker so if the text is huge it doesn't block the JS loop.

I also noticed that we render the full text of the logs into the log rows, even if they're enormous. We only ever show 12 lines with a conditional "Show More", so I added a sanity check that cuts the string down to 15k characters. That should be enough to trigger "Show More", even on a 4000px wide screen, and saves us from computing the bounding box of such enormous blocks of text. Hopefully nobody has been logging like this 😅

image

## How I Tested These Changes

I created a test asset that logged both short and 500k+ text blocks containing links.

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/ui/autolinking.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    range.setEnd(node, (match.index || 0) + match[0].length);
+    const aTag = wrapRangeInNode(range, 'A') as HTMLAnchorElement;
+    aTag.href = href;
+    aTag.rel = 'nofollow noreferrer';
+    aTag.target = '_blank';
+    aTag.title = href;
+  }
+}
+
+export function autolinkTextContent(el: HTMLElement, options: {useIdleCallback: boolean}) {
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/ui/autolinking.ts:181]: Are there any tests available for this as well?

**Review by salazarm — APPROVED** [review]
lgtm, tests would be ideal here :)

File: js_modules/dagster-ui/packages/ui-core/src/ui/autolinking.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
+type AutolinkMatcher = {
+  // A regexp matching the text that should be linked
+  regexp: RegExp;
+  // A prefix that should be prepended to the text to create the link URL.
+  // If the regexp matches ben@dagsterlabs.com, the prefix would be "mailto:".
```
[the remark this answers is not in the document]
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/ui/autolinking.ts:5]: yeah I vote no mailto links for now. This "prefix" feature is in here to support mailto (and tel:, lol), and I didn't remove it, but I think we should just do http:// https:// links unless folks ask for more

File: js_modules/dagster-ui/packages/ui-core/src/ui/autolinking.ts (hunk 1 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/ui/autolinking.ts:181]: Ahh will see if we can pull in some tests, it does seem like it'd be easy to verify a before + after node tree.

**Pull request merged**